File: 4sqstats.rb
require "foursquare2"
require "JSON"

client = Foursquare2::Client.new(:oauth_token => 'ADB02WREAK4W4R5BDYBVEXHWB14VZM4TQOIWZCYAD1GY22EK')

checkins_a = {}

client.user_venue_history().items.each do |v|
	if v.venue.categories[0]
		if v.venue.categories[0].parents
			#puts "#{v.beenHere} #{v.venue.name} #{v.venue.categories[0].parents[0]}"
			checkins_a["#{v.venue.name} - #{v.venue.categories[0].parents[0]}"] = v.beenHere;
		else
			#puts "#{v.beenHere} #{v.venue.name} #{v.venue.categories[0]}"
			checkins_a["#{v.venue.name} - #{v.venue.categories[0]}"] = v.beenHere;
		end
	else
		#puts "#{v.beenHere} #{v.venue.name}"
		checkins_a["#{v.venue.name}"] = v.beenHere;
	end
end

puts checkins_a.sort_by {|k,v| v}.reverse

File: 4sqstats.py
import foursquare

client = foursquare.Foursquare(access_token='ADB02WREAK4W4R5BDYBVEXHWB14VZM4TQOIWZCYAD1GY22EK')
venues = client.users.venuehistory()['venues']['items']

for v in venues:
	if v['venue']['categories']:
		if v['venue']['categories'][0]['shortName']:
			print str(v['beenHere'])+' '+v['venue']['name'].encode('utf-8', 'ignore')+' '+v['venue']['categories'][0]['pluralName'].encode('utf-8', 'ignore')
		else:
			print str(v['beenHere'])+' '+v['venue']['name'].encode('utf-8', 'ignore')+' '+v['venue']['categories'][0].encode('utf-8', 'ignore')
	else:
		print str(v['beenHere'])+' '+v['venue']['name'].encode('utf-8', 'ignore')
